src/Microsoft.DotNet.ProjectModel/ProjectContext.cs

```diff
@@ -146,15 +167,12 @@ public ProjectContext CreateRuntimeContext(IEnumerable<string> runtimeIdentifier
                 return this;
             }
 
-            var standalone = !RootProject.Dependencies
-                .Any(d => d.Type.Equals(LibraryDependencyType.Platform));
-
             var context = CreateBuilder(ProjectFile.ProjectFilePath, TargetFramework)
-                .WithRuntimeIdentifiers(standalone ? runtimeIdentifiers : Enumerable.Empty<string>())
+                .WithRuntimeIdentifiers(runtimeIdentifiers)
                 .WithLockFile(LockFile)
                 .Build();
 
-            if (standalone && context.RuntimeIdentifier == null)
+            if (!context.IsPortable && context.RuntimeIdentifier == null)
```

Comment:
> if !context.IsPortable == context.IsStandalone I would agree.

It is.

You do need to add `runtimes` but that doesn't actually do the work of making it standalone, it's purely the presence of `type:platform` that dictates that, and that's what this property represents.

`!IsPortable && ProjectHasNoRuntimesSection()` would be an error because the project has insufficient data to be published standalone.